src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,156 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value);
+        if (first == null) {
+            first = last = newNode;
+        } else {
+            this.last.next = newNode;
+            last = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index out of range: " + index);
+        }
+        Node<T> newNode = new Node<>(value);
+        if (first == null) {
+            first = last = newNode;
+        } else if (index == 0) {
+            newNode.next = first;
+            first = newNode;
+        } else if (index == size) {
+            last.next = newNode;
+            last = newNode;
+        } else {
+            Node<T> current = findNodeIndex(index - 1);
+            newNode.next = current.next;
+            current.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return findNodeIndex(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> newNode = findNodeIndex(index);
+        T element = newNode.value;
+        newNode.value = value;
+        return element;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        findNodeIndex(index);
+        T deleteElement;
+        if (index == 0) {
+            deleteElement = first.value;
+            first = first.next;
+            if (first == null) {
+                last = null;
+            }
+        } else {
+            Node<T> newNode = findNodeIndex(index - 1);
+            deleteElement = newNode.next.value;
+            newNode.next = newNode.next.next;
+            if (index == size - 1) {
+                last = newNode;
+            }
+        }
+        size--;
+        return deleteElement;
     }
 
     @Override
     public boolean remove(T object) {
+        int index = findIndexObject(object);
+        if (index == 0) {
+            first = first.next;
+            if (first == null) {
+                last = null;
+            }
+            size--;
+            return true;
+        } else if (index > 0 && index < size) {
+            Node<T> newNode = findNodeIndex(index - 1);
+            newNode.next = newNode.next.next;
+            if (index == size - 1) {
+                last = newNode;
+            }
+            size--;
+            return true;
+        }
```

Comment:
You are iterating twice 
the first time in findIndexObject()
the next time in findNodeIndex()

the code is not optimized

also in common mistakes, there was a point about adding method unlink() - to remove nodes

<img width="1130" alt="image" src="https://user-images.githubusercontent.com/81781479/205091585-71514d3f-10da-42f8-afe9-8d6ad43387fa.png">